aws/tagsCognito.go

```diff
@@ -0,0 +1,97 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cognitoidentity"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+)
+
+// setTags is a helper to set the tags for a resource. It expects the
+// tags field to be named "tags"
+func setTagsCognito(conn *cognitoidentity.CognitoIdentity, d *schema.ResourceData) error {
+	if d.HasChange("tags") {
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		create, remove := diffTagsCognito(tagsFromMapCognito(o), tagsFromMapCognito(n))
+
+		// Set tags
+		if len(remove) > 0 {
+			log.Printf("[DEBUG] Removing tags: %s", remove)
+			tagsToRemove := &cognitoidentity.UntagResourceInput{
+				ResourceArn: aws.String(d.Get("arn").(string)),
+				TagKeys:     cognitoStringsToPointyString(remove),
+			}
+
+			_, err := conn.UntagResource(tagsToRemove)
+			if err != nil {
+				return err
+			}
+		}
+		if len(create) > 0 {
+			log.Printf("[DEBUG] Creating tags: %s", create)
+			tagsToAdd := &cognitoidentity.TagResourceInput{
+				ResourceArn: aws.String(d.Get("arn").(string)),
+				Tags:        cognitoTagsFromMap(create),
```

Comment:
I believe this can be replaced with the available AWS Go SDK conversion function [`aws.StringMap()`](https://docs.aws.amazon.com/sdk-for-go/api/aws/#StringMap)

```suggestion
				Tags:        aws.StringMap(create),
```

Reply:
fixed in 4ed6457ff